src/mapkit.cs

```diff
@@ -280,11 +280,14 @@ interface MKDirectionsRequest {
 	[TV (9,2)]
 	[Mac (10,9, onlyOn64 : true)]
 	interface MKMapItem
-#if !MONOMAC
-		: NSSecureCoding
 #if IOS // #if TARGET_OS_IOS
-		, NSItemProviderReading, NSItemProviderWriting
+		: NSItemProviderReading, NSItemProviderWriting
+#if ARCH_64 // [FAIL] Selector not found for MapKit.MKMapItem : encodeWithCoder: on 32-bit iOS
+		, NSSecureCoding
```

Comment:
We can't do that in public API, e.g. if 32bits and 64bits signatures are different then this _leaks_ into user code
It also becomes impossible (or at least harder) to de-duplicate the platform assemblies (for fat apps)

This needs to be considered a OS version change (which it is) e.g. when a new version of iOS adds conformance then we add the API (and fix up introspection not to report it for older versions)

Reply:
My guess is that it's actually tied to the OS version (it works in iOS 11 (64-bit only), but not iOS 10.3 (where we run our 32-bit tests))

Reply:
That explains why I couldn't find many `ARCH_64`, thought I was being clever here :P

But yes I now remember how we usually handle `NSSecureCoding`, I got distracted.